junit-jupiter-params/src/test/java/org/junit/jupiter/params/ParameterizedTestIntegrationTests.java

```diff
@@ -465,18 +469,59 @@ void executesWithEmptySourceForList() {
 			results.testEvents().succeeded().assertEventsMatchExactly(event(test(), displayName("[1] argument=[]")));
 		}
 
+		/**
+		 * @since 5.10
+		 */
+		@ParameterizedTest(name = "{1}")
+		@CsvSource({ //
+				"testWithEmptySourceForArrayList, java.util.ArrayList", //
+				"testWithEmptySourceForLinkedList, java.util.LinkedList"//
+		})
+		void executesWithEmptySourceForListSubtype(String methodName, Class<?> parameterType) {
+			var results = execute(methodName, parameterType);
+			results.testEvents().succeeded().assertEventsMatchExactly(event(test(), displayName("[1] argument={}")));
+		}
```

Comment:
For some reason, `executesWithEmptySourceForMapSubtype` passes but `executesWithEmptySourceForListSubtype` and `executesWithEmptySourceForSetSubtype` fail. Also, going with separate tests work correctly.

Am I trying to do something that is not supposed to work?

Reply:
It seems you figured it out?

Reply:
Not really, I went back to the single test approach purely to move forward but I plan to give another try to parameterized tests once I'm done with everything else.

Reply:
Not sure what I've done wrong before... but seems to work fine now 🙂